tmt/steps/report/reportportal.py

```diff
@@ -15,49 +16,95 @@ class ReportReportPortalData(tmt.steps.report.ReportStepData):
         help="The launch name (name of plan per launch is used by default).")
+    launch_description: str = field(
+        option="--launch-description",
+        metavar="LAUNCH_DESCRIPTION",
+        default=os.getenv('TMT_PLUGIN_REPORT_REPORTPORTAL_LAUNCH_DESCRIPTION'),
+        help="Pass the description for ReportPortal launch, especially with '--suite-per-plan' "
+             "option (Otherwise Summary from plan fmf data per each launch is used by default).")
+    launch_per_plan: bool = field(
+        option="--launch-per-plan",
+        default=False,
+        is_flag=True,
+        help="Mapping launch per plan, creating one or more launches with no suite structure.")
+    suite_per_plan: bool = field(
+        option="--suite-per-plan",
+        default=False,
+        is_flag=True,
+        help="Mapping suite per plan, creating one launch and continous uploading suites into it. "
+             "Recommended to use with '--launch' and '--launch-description' options."
+             "Can be used with '--upload-to-launch' option to avoid creating a new launch.")
+    upload_to_launch: str = field(
+        option="--upload-to-launch",
+        metavar="LAUNCH_ID",
+        default=None,
+        help="Pass the launch ID for an additional test/suite upload to an existing launch. "
+             "ID can be found in the launch URL.")
+    upload_to_suite: str = field(
+        option="--upload-to-suite",
+        metavar="LAUNCH_SUITE",
+        default=None,
+        help="Pass the suite ID for an additional test upload to an existing launch. "
+             "ID can be found in the suite URL.")
+    launch_rerun: bool = field(
+        option="--launch-rerun",
+        default=False,
+        is_flag=True,
+        help="Rerun the launch based on unique test paths and ids to create Retry item"
+             "with a new version per each test. Supported in 'suite-per-plan' structure only.")
+    defect_type: Optional[str] = field(
+        option="--defect-type",
+        metavar="DEFECT_NAME",
+        default=None,
+        help="Pass the defect type to be used for failed test, which is defined in the project"
+             " (e.g. 'Idle'). 'To Investigate' is used by default.")
+    # TODO: test how to create empty test skeleton, all with Idle defect_type
+    # (as it reports defect_type only when it fails)
     exclude_variables: str = field(
         option="--exclude-variables",
         metavar="PATTERN",
         default="^TMT_.*",
-        help="""
-             Regular expression for excluding environment variables from reporting to ReportPortal
-             ('^TMT_.*' used by default).
-             """)
+        help="Regular expression for excluding environment variables "
+             "from reporting to ReportPortal ('^TMT_.*' used by default).")
+    launch_url: str = ""
+    launch_uuid: str = ""
+    suite_uuid: str = ""
+    test_uuids: list[str] = ""
```

Comment:
Four weird-looking fields :)

* unless an empty string is really a valid launch URL/launch UUID/suite UUID, these three should be `Optional[str] = None`. `None` serves better as an expression of "not set", way better than an empty string which is much harder to follow for linters.
* `list[str] = ""` will not be accepted by linter, and either the type is incorrect, or the initial value is.